libraries/Microsoft.Bot.Connector/Authentication/MicrosoftGovernmentAppCredentials.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    /// <summary>
+    /// MicrosoftGovernmentAppCredentials auth implementation
+    /// </summary>
+    public class MicrosoftGovernmentAppCredentials : MicrosoftAppCredentials
+    {
+        /// <summary>
+        /// An empty set of credentials.
+        /// </summary>
+        public static new readonly MicrosoftGovernmentAppCredentials Empty = new MicrosoftGovernmentAppCredentials(null, null);
+        
+        static MicrosoftGovernmentAppCredentials()
+        {
```

Comment:
As a general rule type initializers are best avoided, as it's extremely non-deterministic when they run. 

Please strongly consider removing this type initializer and doing this a different way. 

Jon Skeet uses these as brain-teasers as the behavior is so wildly unpredictable:
[https://www.google.com/search?q=jon+skeet+type+initializer](https://www.google.com/search?q=jon+skeet+type+initializer)

Reply:
I got rid of these and just added them to the base class. There is no harm in adding to our "known good" list.